addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/handler/SatelThingHandler.java

```diff
@@ -0,0 +1,179 @@
+            }
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } else if (event instanceof IntegraStateEvent) {
+            // update thing's state unless it should accept commands only
+            IntegraStateEvent stateEvent = (IntegraStateEvent) event;
+            if (thingConfig.isCommandOnly()) {
+                return;
+            }
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                StateType stateType = getStateType(channelUID.getId());
+                if (stateType != null && stateEvent.hasDataForState(stateType)) {
+                    int bitNbr = thingConfig.getId() - 1;
+                    boolean invertState = thingConfig.isStateInverted();
+                    updateSwitch(channelUID, stateEvent.isSet(stateType, bitNbr) ^ invertState);
+                }
+            }
+        }
+    }
+
+    protected abstract SatelCommand convertCommand(ChannelUID channel, Command command);
+
+    protected abstract StateType getStateType(String channelId);
+
+    protected Channel getChannel(StateType stateType) {
+        String channelId = stateType.toString().toLowerCase();
+        Channel channel = getThing().getChannel(channelId);
+        if (channel == null) {
+            logger.warn("Missing channel for {}", stateType);
+        }
+        return channel;
+    }
+
+    protected Collection<SatelCommand> getRefreshCommands(NewStatesEvent event) {
+        Collection<SatelCommand> result = new LinkedList<>();
+        boolean requiresRefresh = requiresRefresh();
+        for (Channel channel : getThing().getChannels()) {
+            boolean isChannelLinked = !linkRegistry.getLinks(channel.getUID()).isEmpty();
+            if (isChannelLinked) {
+                StateType stateType = getStateType(channel.getUID().getId());
+                if (requiresRefresh || event.isNew(stateType.getRefreshCommand())) {
+                    result.add(new IntegraStateCommand(stateType, bridgeHandler.getIntegraType().hasExtPayload()));
+                }
+            }
+        }
+        return result;
+    }
+
+    protected void updateSwitch(String channelID, boolean switchOn) {
+        ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), channelID);
+        updateSwitch(channelUID, switchOn);
+    }
+
+    protected void updateSwitch(ChannelUID channelUID, boolean switchOn) {
+        State state = switchOn ? OnOffType.ON : OnOffType.OFF;
+        updateState(channelUID, state);
+    }
+
+    protected byte[] getObjectBitset(int size, int... ids) {
+        byte[] bitset = new byte[size];
+        for (int id : ids) {
+            int bitNbr = id - 1;
+            bitset[bitNbr / 8] |= (byte) (1 << (bitNbr % 8));
+        }
+        return bitset;
+    }
+
+    protected boolean requiresRefresh() {
```

Comment:
Not clear what this method is used for exactly and how it works.
Note that you should never really come across INITIALIZING Thing status during the lifetime of your handler.

Reply:
If the code is here, that means it **did** happen. Not sure if still can.

Reply:
If it did happen, than only due to the fact that you miss to set the Thing status in the initialize() method - you need to set it to either UNKNOWN, ONLINE and OFFLINE there.

Reply:
Done.